core/src/main/scala/kafka/coordinator/transaction/TransactionStateManager.scala

```diff
@@ -271,31 +272,22 @@ class TransactionStateManager(brokerId: Int,
    * When this broker becomes a follower for a transaction log partition, clear out the cache for corresponding transactional ids
    * that belong to that partition.
    */
-  def removeTransactionsForPartition(partition: Int) {
+  def removeTransactionsForTxnTopicPartition(partitionId: Int) {
     validateTransactionTopicPartitionCountIsStable()
 
-    val topicPartition = new TopicPartition(Topic.TransactionStateTopicName, partition)
-
-    inLock(stateLock) {
-      ownedPartitions.remove(partition)
-      loadingPartitions.remove(partition)
-    }
+    val topicPartition = new TopicPartition(Topic.TransactionStateTopicName, partitionId)
```

Comment:
We have the situation here that when this method returns nothing has actually happened yet, i.e., the removal is done on a different thread. So what happens if we get a request at the same time for another transactionalId that is in the same partition? Are we just relying/hoping it will eventually fail?